src/pages/ScholarshipForm.jsx

```diff
@@ -1,21 +1,112 @@
 import React, { useState } from 'react';
 import { Container, Button, TextField } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
 import ScholarshipAmountField from '../components/ScholarshipAmountField';
 import DatePicker from '../components/DatePicker';
+import AmountType from '../types/AmountType';
+
+const useStyles = makeStyles({
+  textFieldStyle: {
+    minWidth: 400,
+  },
+});
+
+// regular expression used for valid website urls
+const websiteReg = /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([-.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/;
```

Comment:
I recommend trying these out on a regex testing website like https://regexr.com/ and testing to see if there's a simpler form that's still relatively simple to read. E.g. `(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?` can be simplified to `(https?:\/\/(www\.)?)?`. It captures the optional `s` and the optional `www.`. If you want to show the optional `s` a little more clearly you could do `(http(s)?:\/\/(www\.)?)?`.

```suggestion
const websiteReg = /^(http(s)?:\/\/(www\.)?)?[\w-]+(\.[\w-]+)*\.[a-z]{2,}((\/|#|\?)(\S)*)?$/;
```

You can play around with it [here](https://regexr.com/5g91f)

Reply:
Much simpler!